app/services/auth.js

```diff
@@ -55,20 +56,29 @@ export default Service.extend({
     this.store.unloadAll();
   },
 
-  signIn(data, options = {}) {
+  signInWith(provider, data, options = {}) {
+    if (availableProviders.includes(provider)) {
+      this.signIn(data, options, provider);
+    } else {
+      throw new Error(`Invalid provider to authenticate ${provider}`);
+    }
```

Comment:
You can use [`assert`](https://api.emberjs.com/ember/3.12/functions/@ember%2Fdebug/assert) here instead:
```js
assert(`Invalid provider to authenticate ${provider}`, availableProviders.includes(provider));
this.signIn(data, options, provider);
```